internal/monitoring/request.go

```diff
@@ -7,27 +7,55 @@ import (
 )
 
 // Request represents a monitoring request.
+// The zero value is good to use after calling Reset.
 type Request struct {
 	startTime time.Time
-	label     string
+	root      string
+	method    string
+	kind      string
 }
 
-// Reset the monitor start time to now and the label to blank.
-func (r *Request) Reset() {
+var knownMethods = map[string]bool{
+	"GET":     true,
+	"POST":    true,
+	"PUT":     true,
+	"HEAD":    true,
+	"OPTIONS": true,
+}
+
+// NewRequest returns a new monitoring request
+// for monitoring a request within the given root.
+// When the request is done, Done should be called.
+func NewRequest(method, root string) *Request {
+	var req Request
+	req.Reset(method, root)
+	return &req
+}
+
+// Reset resets r to indicate that a new request has started. The
+// parameter holds the API root (for example the API version).
```

Comment:
So root is like a prefix? Correct? And kind is like specializing requests done to the same root.